examples/platforms/simulation/system.c

```diff
@@ -127,6 +130,9 @@ void otSysInit(int aArgCount, char *aArgVector[])
         case OT_SIM_OPT_HELP:
             PrintUsage(aArgVector[0], EXIT_SUCCESS);
             break;
+        case OT_SIM_OPT_ENABLE_ENERGY_SCAN:
+            gRadioCaps |= OT_RADIO_CAPS_ENERGY_SCAN;
```

Comment:
Hi Yakun, what is the background/use-case of adding energy scan support in radio platform? Why do we need to have energy scan in simulation?

One of the key goals of simulation platform is to verify and validate the implementation in OT core. So it make sense to not implement and include radio capabilities in simulation radio itself and leave it to OT core implementation to be used (so we validate the implementation in core in `SubMac`/`Mac`).

`scan energy` does work now and it uses the implementation in `SubMac` along with `otPlatRadioGetRssi()`. If you want to make it look more random, maybe can consider changing the `otPlatRadioGetRssi()` code:

```c++
int8_t otPlatRadioGetRssi(otInstance *aInstance)
{
    OT_UNUSED_VARIABLE(aInstance);

    assert(aInstance != NULL);

    int8_t   rssi    = SIM_LOW_RSSI_SAMPLE;
    uint8_t  channel = sReceiveFrame.mChannel;
    uint32_t probabilityThreshold;

    otEXPECT((SIM_RADIO_CHANNEL_MIN <= channel) && channel <= (SIM_RADIO_CHANNEL_MAX));

    // To emulate a simple interference model, we return either a high or
    // a low  RSSI value with a fixed probability per each channel. The
    // probability is increased per channel by a constant.

    probabilityThreshold = (channel - SIM_RADIO_CHANNEL_MIN) * SIM_HIGH_RSSI_PROB_INC_PER_CHANNEL;

    if (otRandomNonCryptoGetUint16() < (probabilityThreshold * 0xffff / 100))
    {
        rssi = SIM_HIGH_RSSI_SAMPLE;
    }

exit:
    return rssi;
}
```

This code now returns either a high (-30) or low (-98) value randomly based on the channel.
- There is pattern which ensures on average lower channel numbers look better (less noisy) which is what is assumed/relied upon in some of the test cases around "channel monitor" and "channel manager" (which use simulation platform).
- We can add randomness (pick a value within a high range or low range) based on channel number if the goal is to make the output of `scan energy` more random.

Reply:
We encountered a bug on posix platform. The bug is related to RCP with energy scan capability which causes calling platform radio APIs directly in the energy scan done callback. The bug is not easy to reproduce on a real device, so I want to add this implementation to help reproduce the bug and verify it's fix.

Reply:
Updated to share the RSSI simulation.